enhancements/security/file-integrity-operator.md

```diff
@@ -0,0 +1,30 @@
+# Cluster Node File Integrity Operator
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift/docs]
+
+## Open Questions [optional]
+
+## Summary
+
+This enhancement describes a new security feature for OpenShift. Many security-conscious customers want to be informed when files on a host's filesystem are modified in a way that is unexpected, as this may indicate an attack or compromise.  It proposes a "file-integrity-operator" that provides file integrity monitoring of select files on the host filesystems of the cluster nodes. It periodically runs a verification check on the watched files and provides logs of any changes.
+
+## Motivation
+
+In addition to the reasons stated in the Summary section, as part of the FedRAMP gap assessment of OpenShift/CoreOS, it has been identified that to fulfill several NIST SP800-53 security controls we need to constantly do integrity checks on configuration files (CM-3 & CM-6), as well as critical system paths and binaries (boot configuration, drivers, firmware, libraries) (SI-7). Besides verifying the files, we need to be able to report which files changed and in what manner, in order for the organization to better determine if the change has been authorized or not. In order to fulfull the controls the file integrity checks need to be done using a state-of-the-practice integrity checking mechanism (e.g., parity checks, cyclical redundancy checks, cryptographic hashes). If using cryptographic hashes for integrity checks, such algorithms need to be FIPS-approved.
```

Comment:
s/CoreOS/RHCOS/

But it is unclear whether:
 * this applies to RHCOS only, because on other OSes you expect some other solutions to cover the same usecase (which? how?)
 * this applies to RHCOS only, because on other OSes we don't want cluster-orchestrated file integrity monitoring
 * this is not specific to RHCOS, but applies to any OS where the OpenShift cluster is running

Reply:
Thanks, this is not specific to RHCOS but for any type of node.

Reply:
Ack. Then I think you can simply drop RHCOS at all from here, as the gap you are trying to address is not specific to it.